src/phyltr/commands/base.py

```diff
@@ -115,20 +115,22 @@ def post_print(self):
     # The conceptual heart of phyltr...
 
     def consume(self, stream):
-        for tree in stream:
+        i = 0
+        for i, tree in enumerate(stream, start=1):
             try:
-                res = self.process_tree(tree)
+                res = self.process_tree(tree, i)
                 if res is not None:
                     yield res
             except StopIteration:
                 if hasattr(stream, 'close'):
                     stream.close()
                 break
-        for tree in self.postprocess():
+        for tree in self.postprocess(i):
             yield tree
 
-    def process_tree(self, t):
-        return t    # pragma: no cover
+    def process_tree(self, t, n):
```

Comment:
Ah, this (building enumeration into the base command and ignoring it when not needed) makes a lot of sense!  Nice idea.

Reply:
Initially I wanted to consolidate the various `self.n` and `self.N` attributes into a `PhyltrCommand.tree_count`, but then found less state even more elegant.